add_glyphs.py

```diff
@@ -0,0 +1,403 @@
+    if seq in seq_to_file:
+      raise Exception('duplicate sequence for "%s" in %s' % (name, image_dir))
+    seq_to_file[seq] = path.join(image_dir, name)
+  return seq_to_file
+
+
+def collect_seq_to_file(image_dirs, prefix, suffix):
+  """Return a sequence to file mapping by calling get_seq_to_file on a list
+  of directories.  When sequences for files in later directories match those
+  from earlier directories, the later file replaces the earlier one.
+  """
+  seq_to_file = {}
+  for image_dir in image_dirs:
+    seq_to_file.update(get_seq_to_file(image_dir, prefix, suffix))
+  return seq_to_file
+
+
+def remap_values(seq_to_file, map_fn):
+  return {k: map_fn(v) for k, v in seq_to_file.iteritems()}
+
+
+def get_png_file_to_advance_mapper(lineheight):
+  def map_fn(filename):
+    wid, ht = PNG(filename).get_size()
+    return int(round(float(lineheight) * wid / ht))
+  return map_fn
+
+
+def cp_name(cp):
+  """return uniXXXX or uXXXXX(X) as a name for the glyph mapped to this cp."""
+  return '%s%04X' % ('u' if cp > 0xffff else 'uni', cp)
+
+
+def seq_name(seq):
+  """Sequences of length one get the cp_name.  Others start with 'u' followed by
+  two or more 4-to-6-digit hex strings separated by underscore."""
+  if len(seq) == 1:
+    return cp_name(seq[0])
+  return 'u' + '_'.join('%04X' % cp for cp in seq)
+
+
+def collect_cps(seqs):
+  cps = set()
+  for seq in seqs:
+    cps.update(seq)
+  return cps
+
+
+def get_glyphorder_cps_and_truncate(glyphOrder):
+  """This scans glyphOrder for names that correspond to a single codepoint
+  using the 'u(ni)XXXXXX' syntax.  All names that don't match are moved
+  to the front the glyphOrder list in their original order, and the
+  list is truncated.  The ones that do match are returned as a set of
+  codepoints."""
+  glyph_name_re = re.compile(r'^u(?:ni)?([0-9a-fA-F]{4,6})$')
+  cps = set()
+  write_ix = 0
+  for ix, name in enumerate(glyphOrder):
+    m = glyph_name_re.match(name)
+    if m:
+      cps.add(int(m.group(1), 16))
+    else:
+      glyphOrder[write_ix] = name
+      write_ix += 1
+  del glyphOrder[write_ix:]
+  return cps
+
+
+def get_all_seqs(font, seq_to_advance):
+  """Copies the sequences from seq_to_advance and extends it with single-
+  codepoint sequences from the GlyphOrder table as well as those internal
+  to sequences in seq_to_advance.  Reduces the GlyphOrder table. """
+
+  all_seqs = set(seq_to_advance.keys())
+  # using collect_cps includes cps internal to a seq
+  cps = collect_cps(all_seqs)
+  glyphOrder = font.getGlyphOrder()
+  # extract cps in glyphOrder and reduce glyphOrder to only those that remain
+  glyphOrder_cps = get_glyphorder_cps_and_truncate(glyphOrder)
+  cps.update(glyphOrder_cps)
+  # add new single codepoint sequences from glyphOrder and sequences
+  all_seqs.update((cp,) for cp in cps)
+  return all_seqs
+
+
+def get_font_cmap(font):
+  """Return the first cmap in the font, we assume it exists and is a unicode
+  cmap."""
```

Comment:
The template file includes the right cmap.  This is the behavior of the old add_glyphs.py file, I just added a getter for it since we do this in a few places instead of putting it into a global variable and reusing it all over.

An exception will get thrown eventually :-)  I might verify this when I next change the file.